// ngx-extended-pdf-viewer/esm2022/lib/translate.pipe.mjs
import { effect, Pipe } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./pdf-notification-service";
export class TranslatePipe {
    PDFViewerApplication;
    constructor(notificationService) {
        effect(() => {
            this.PDFViewerApplication = notificationService.onPDFJSInitSignal();
        });
    }
    transform(key, fallback) {
        return this.translate(key, fallback);
    }
    async translate(key, englishText) {
        while (!this.PDFViewerApplication) {
            console.log('waiting for PDFViewerApplication to translate ' + key);
            await new Promise((resolve) => setTimeout(resolve, 1));
        }
        return this.PDFViewerApplication?.l10n.get(key, null, englishText);
    }
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: TranslatePipe, deps: [{ token: i1.PDFNotificationService }], target: i0.ɵɵFactoryTarget.Pipe });
    static ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "17.3.12", ngImport: i0, type: TranslatePipe, name: "translate" });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: TranslatePipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'translate',
                }]
        }], ctorParameters: () => [{ type: i1.PDFNotificationService }] });
