// powerbi-client-angular/esm2022/powerbi-embed.module.mjs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { NgModule } from '@angular/core';
import { PowerBIEmbedComponent } from './components/powerbi-embed/powerbi-embed.component';
import { PowerBIDashboardEmbedComponent } from './components/powerbi-dashboard-embed/powerbi-dashboard-embed.component';
import { PowerBIPaginatedReportEmbedComponent } from './components/powerbi-paginated-report-embed/powerbi-paginated-report-embed.component';
import { PowerBIQnaEmbedComponent } from './components/powerbi-qna-embed/powerbi-qna-embed.component';
import { PowerBIReportEmbedComponent } from './components/powerbi-report-embed/powerbi-report-embed.component';
import { PowerBITileEmbedComponent } from './components/powerbi-tile-embed/powerbi-tile-embed.component';
import { PowerBIVisualEmbedComponent } from './components/powerbi-visual-embed/powerbi-visual-embed.component';
import { PowerBICreateReportEmbedComponent } from './components/powerbi-create-report/powerbi-create-report.component';
import * as i0 from "@angular/core";
export class PowerBIEmbedModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: PowerBIEmbedModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.3.12", ngImport: i0, type: PowerBIEmbedModule, declarations: [PowerBIEmbedComponent,
            PowerBIDashboardEmbedComponent,
            PowerBIPaginatedReportEmbedComponent,
            PowerBIQnaEmbedComponent,
            PowerBIReportEmbedComponent,
            PowerBITileEmbedComponent,
            PowerBIVisualEmbedComponent,
            PowerBICreateReportEmbedComponent], exports: [PowerBIDashboardEmbedComponent,
            PowerBIPaginatedReportEmbedComponent,
            PowerBIQnaEmbedComponent,
            PowerBIReportEmbedComponent,
            PowerBITileEmbedComponent,
            PowerBIVisualEmbedComponent,
            PowerBICreateReportEmbedComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: PowerBIEmbedModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: PowerBIEmbedModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        PowerBIEmbedComponent,
                        PowerBIDashboardEmbedComponent,
                        PowerBIPaginatedReportEmbedComponent,
                        PowerBIQnaEmbedComponent,
                        PowerBIReportEmbedComponent,
                        PowerBITileEmbedComponent,
                        PowerBIVisualEmbedComponent,
                        PowerBICreateReportEmbedComponent
                    ],
                    imports: [],
                    exports: [
                        PowerBIDashboardEmbedComponent,
                        PowerBIPaginatedReportEmbedComponent,
                        PowerBIQnaEmbedComponent,
                        PowerBIReportEmbedComponent,
                        PowerBITileEmbedComponent,
                        PowerBIVisualEmbedComponent,
                        PowerBICreateReportEmbedComponent
                    ],
                }]
        }] });
